server/src/main/java/org/candlepin/model/OwnerCurator.java

```diff
@@ -47,6 +48,36 @@ public OwnerCurator() {
         super(Owner.class);
     }
 
+    /**
+     * Fetches the Owner for the specified ownerId. If the ownerId is null or owner was not found, this
+     * method throws an exception.
+     *
+     * @param ownerId
+     *  The ownerId for which to fetch a Owner object
+     *
+     * @throws IllegalArgumentException
+     *  if ownerId is null
+     *
+     * @throws IllegalStateException
+     *  if the owner was not found for the given id
+     *
+     * @return
+     *  An Owner instance for the specified ownerId
+     */
+    public Owner findOwnerById(String ownerId) {
```

Comment:
@Ceiu  is this a suggestion or a more stricter request for change? I prefer using the owner id instead of another object cause that way its generic and can be used in more use cases irrespective of whether the owner id comes from a consumer or an owned object.

Reply:
why does it have to come from a persisted object? why should the method implementation care about the context in which the owner is being looked up? I dont see the harm in having a method that looks up an owner with owner id.

Reply:
Exactly for the reason you just stated. If we're passing in just a string, there should be no assumption of its source and, thus, no expectation of a given state. With such a method, doing state management is offloading work the caller should be doing. 

When we pass in an object from which we pull the ID, the method can now make assumptions about state of the ID value: non-null and valid. At this point, it's fair to be doing state management.

And, again, we already have ID-based lookups for owner ID.

Reply:
are you refering to find / get?

Reply:
Yes; especially if/when we get around to fixing our convention to match up with Hibernate's use of find/get.

Reply:
okay. I first implemented this PR while using find, but as you pointed out there is no validation, so I added this method which is basically find+validation.

Reply:
Right; the intent there was to call out that we need to either (a) add another method like the getConsumerType that validates the object's state while doing the lookup, or (b) add validation to the output of the calls to .find.

Reply:
im okay with another method like find with added validation to it, imo its better than (b) , and I dont quite like (a) cause that would mean one method for each object type.
is this is a suggestion or a more stricter request for change?

Reply:
>  dont see the harm in having a method that looks up an owner with owner id.

+1

I don't buy arguments about the method making assumptions about state, I'd rather have employee defensive programming practices and do a check.

Let's stick with findOwnerById(String) as written.

> In the next scrum and we'll discuss it with the team.

I'm happy to discuss and revisit with convincing arguments :smile:, for now, what is here is fine. (We can always revisit outside the scope of this PR).

Reply:
You and I both know this will never be changed, and quite frankly, I'm sick of our conventions changing between curators and objects without any logic because "I felt like doing it this way"

Maintaining CP is a bear already, and every time we decide to do something differently on a whim is just raising that bar and I'm growing quite tired of seeing patchwork things go through without any value to justify the complete deviation from other methods. But whatever, what's another such patch when we already have a plethora of others, I suppose. :/